src/v/kafka/client/consumer.cc

```diff
@@ -104,6 +104,9 @@ ss::future<> consumer::join() {
               return join();
           case error_code::illegal_generation:
               return join();
+          case error_code::not_coordinator:
+              return ss::sleep(shard_local_cfg().retry_base_backoff())
```

Comment:
use abortable sleep? perhaps it doesn't matter if the backoff is very short, but i don't have a good idea of this config value upper bound.

Reply:
I have a separate task for properly threading cancellation.

Reply:
how do you cancel a sleep?

Reply:
Added an abort source.